src/NimBLEDevice.cpp

```diff
@@ -705,6 +707,36 @@ void NimBLEDevice::setSecurityCallbacks(NimBLESecurityCallbacks* callbacks) {
 } // setSecurityCallbacks
 
 
+/**
+ * @brief Set the own address type.
+ * @param own_addr_type Own Bluetooth Device address type.\n
+ * The available bits are defined as:
+ * * 0x00: BLE_OWN_ADDR_PUBLIC
+ * * 0x01: BLE_OWN_ADDR_RANDOM
+ * * 0x02: BLE_OWN_ADDR_RPA_PUBLIC_DEFAULT
+ * * 0x03: BLE_OWN_ADDR_RPA_RANDOM_DEFAULT
+ */
+void NimBLEDevice::setOwnAddrType(uint8_t own_addr_type, bool useNRPA) {
+    m_own_addr_type = own_addr_type;
+    if (own_addr_type != BLE_OWN_ADDR_PUBLIC) {
+        setSecurityInitKey(BLE_SM_PAIR_KEY_DIST_ID);
```

Comment:
This needs to be `setSecurityInitKey(BLE_SM_PAIR_KEY_DIST_ENC | BLE_SM_PAIR_KEY_DIST_ID);` or we won't be able to encrypt connections.